pqarrow/arrow.go

```diff
@@ -82,10 +73,142 @@ func rowBasedParquetRowGroupToArrowRecord(
 		}
 	}
 
+	return arrow.NewSchema(fields, nil), nil
+}
+
+// contiguousParquetRowGroupToArrowRecord converts a parquet row group to an arrow record.
+func contiguousParquetRowGroupToArrowSchema(
+	ctx context.Context,
+	pool memory.Allocator,
+	rg parquet.RowGroup,
+	projections []logicalplan.ColumnMatcher,
+	filterExpr logicalplan.Expr,
+	distinctColumns []logicalplan.ColumnMatcher,
+) (*arrow.Schema, error) {
+	parquetColumns := rg.ColumnChunks()
+	parquetFields := rg.Schema().Fields()
+
+	if len(distinctColumns) == 1 && filterExpr == nil {
+		// We can use the faster path for a single distinct column by just
+		// returning its dictionary.
+		fields := make([]arrow.Field, 0, 1)
+		for i, field := range parquetFields {
+			select {
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			default:
+				name := field.Name()
+				if distinctColumns[0].Match(name) {
+					typ, nullable, _, err := parquetColumnToArrowArray(
```

Comment:
doing the entire array conversion seems awfully expensive just to get the schema